pkgs/by-name/xr/xrsh/package.nix

```
@@ -0,0 +1,62 @@
+{
+  stdenv,
+  fetchFromGitea,
+  makeWrapper,
+  writeShellApplication,
+  python3Packages,
+  lib,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xrsh";
+  version = "0.143";
+
+  src = fetchFromGitea {
+    fetchSubmodules = true;
+    domain = "codeberg.org";
+    owner = "xrsh";
+    repo = "xrsh";
+    url = "https://codeberg.org/xrsh/xrsh.git";
+    rev = "435ac2fc51d5966a86607bb5ede3661a210157d4";
+    hash = "sha256-g+YxeExomlWHF7suzlPjUlZe6eXa9MPA10bCoifQO1I=";
+  };
+
+  dontBuild = true;
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase =
+    let
+      launcher = writeShellApplication {
+        name = "xrsh-launcher";
+        runtimeInputs = [ python3Packages.twisted ];
+        text = ''
+          XRSH_ROOT="''${1}"
+          XRSH_PORT="''${XRSH_PORT-8080}"
+
+          exec twistd -n web --listen "tcp:''${XRSH_PORT}" --path "''${XRSH_ROOT}"
+        '';
+      };
+    in
+    ''
+      runHook preInstall
+
+      mkdir -p $out/{bin,share/xrsh}
+      cp -r -t $out/share/xrsh index.html src xrsh.ico xrsh.svg
+
+      makeWrapper ${lib.getExe launcher} $out/bin/xrsh
```

Don't forget to break the line ;)

```suggestion
      makeWrapper ${lib.getExe launcher} $out/bin/xrsh \
```